spec/60-trace-id-format.md

```diff
@@ -1,20 +1,52 @@
-the system works with an 8-byte `trace-id` like `3ce929d0e0e4736`. Instead of
-setting `trace-id` value to `0000000000000003ce929d0e0e4736` it SHOULD generate
-a value like `4bf92f3577b34da6a3ce929d0e0e4736` where `4bf92f3577b34da6a` is a
-random value or a function of time and host value.
-
-**Note**: Even though a system may operate with a shorter `trace-id` for
-[distributed trace](https://w3c.github.io/trace-context/#dfn-distributed-traces)
-reporting, the full `trace-id` MUST be propagated to conform to the
-specification.
+This section suggests some best practices to consider when platform or tracing
+vendor implement `trace-id` generation and propagation algorithms. These
+practices will ensure better interoperability of different systems.
+
+### Uniqueness of `trace-id`
+
+The value of `trace-id` SHOULD be globally unique. This field is typically used
+for unique identification of a <a>distributed trace</a>. It is common for
+<a>distributed traces</a> to span various components, including, for example,
+cloud services. Cloud services tend to serve variety of clients and have a very
+high throughput of requests. So global uniqueness of `trace-id` is important,
+even when local uniqueness might seem like a good solution.
+
+### Randomness of `trace-id`
+
+Randomly generated value of `trace-id` SDHOULD be preferred over other
+algorithms of generating a globally unique identifiers. Randomness of `trace-id`
+addresses some [secutiry](#security-considerations) and [privacy
+concerns](#privacy-considerations) of exposing unwanted information. Randomness
+also allows tracing vendors to base sampling decisions on `trace-id` field value
+and avoid propagating an additional sampling context.
+
+As shown in the next section, it is important for better inter-operability with
+some existing systems, for `trace-id` to carry "uniqueness" and "randomness" in
+the right part of the `trace-id`.
+
+### Length of `trace-id` for the platforms operating with shorter identifiers
+
+There are system operates with a `trace-id` that is shorter than 16 bytes,
+which still willing to adopt this specification.
+
+If such system capable of propagating an entire `trace-id`, even while operating
+internally on shorter identifier, system is encouraged to utilize the
+`tracestate` header to propagate shorter identifier. However, if system wants to
+use a subset of trace-id bytes as an identifier - rightmost part of a `trace-id`
+SHOULD be used.
+
+There are tracing systems that are not capable to propagate the entire 16 bytes
+of a `trace-id`. For better interoperability of these systems with the fully
+compliant systems, the following practices are recommended:
+
+1. Whenever a new `trace-id` needs to be generated - it SHOULD be left padded
+   with zeroes. For example, for the identifier `53ce929d0e0e4736`, `trace-id`
+   must be set to `000000000000000053ce929d0e0e4736`.
+2. On incoming `traceparent`, the rightmost part of `trace-id` SHOULD be used as
```

Comment:
I would split this into two bullet items:
  * on ingress, the incoming trace-if should be truncated by dropping the left-most bytes
  * on egress, the outgoing trace-id should be formed by padding the internal 64bit id with 0 bytes on the left

Reply:
I agree about reframing this section, but I don't see it so much as ingress vs egress, I see it as crossing W3C compliance boundaries.  

When a trace moves from a 64 bit system to a W3C compliant 128 bit system, the 128 bit system should left-pad with zeros.  

When a trace moves from a 128 bit system to a 64 bit system, we expect the 64 bit system to either propagate the full 128 bits (if possible) or propagate only the 64 least-significant bits of the trace id.

Reply:
Just to add a further clarification: the conversions must happen in the new, complaint systems, not in the legacy systems. The legacy systems should be assumed to be unchangeable. So the egress/ingress directions are slightly different.

I also feel that it would be clearer to have two completely separate sections: one for describing new systems which use a non-complaint internal identifier, and one for new systems interoperating with legacy systems.

This would be my suggested change:

```markdown
### Handling `trace-id` for complaint platforms with shorter internal identifiers

There are tracing systems which use a `trace-id` that is shorter than 16 bytes, 
which are still willing to adopt this specification.

If such a system is capable of propagating a fully compliant `trace-id`, even 
while still requiring a shorter, non-complaint identifier for internal purposes, 
the system is encouraged to utilize the `tracestate` header to propagate the 
additional internal identifier. However, if a system would instead prefer to use 
the internal identifier as the basis for a fully complaint `trace-id`, it SHOULD 
be incorporated at the as rightmost part of a `trace-id`.

### Interoperating with legacy formats which use shorter identifiers

There are tracing systems which are not capable of propagating the entire 16 
bytes of a `trace-id`. When a fully compliant system wished to interoperarate 
with these legacy systems, the following practices are recommended:

1. Whenever a shorter `trace-id` is recieved from a legacy system, a compliant 
   `trace-id` SHOULD be generated by left padding the original identifier with 
   zeroes. For example, when receiving the identifier `53ce929d0e0e4736`, the 
   new `trace-id` should be set to `000000000000000053ce929d0e0e4736`.
2. When converting `traceparent` to a legacy format, the rightmost part of 
   `trace-id` SHOULD be used as the identifier when truncating. For instance, 
   if the the value of `trace-id` was `234a5bcd543ef3fa53ce929d0e0e4736` on an 
   incoming request, all outgoing requests to legacy systems would be sent a 
   `trace-id` with the value of `53ce929d0e0e4736`.
```

Reply:
I think I resolved it https://github.com/w3c/trace-context/pull/356/commits/75f04af39c509ba061d9c64e2979234dfe2274a6

Modified text suggested by @tedsuo a bit. And added a note about `tracestate` propagation